src/baal/utils/metrics.py

```diff
@@ -1,10 +1,28 @@
 # From https://github.com/pytorch/ignite with slight changes
+import dataclasses
 import math
 import warnings
+from collections import defaultdict
 
 import numpy as np
 import torch
-from sklearn.metrics import confusion_matrix
+from sklearn.metrics import confusion_matrix, auc
+
+
+def to_prob(probabilities):
+    """
+    If the probabilities tensor is not a distrubution will softmax it.
+
+    Args:
+        probabilities (tensor): [batch_size, num_classes, ...]
+
+    Returns:
+        Same as probabilities.
+    """
+    bounded = torch.min(probabilities) < 0 or torch.max(probabilities) > 1.0
+    if bounded or not torch.allclose(probabilities.sum(1), torch.ones_like(probabilities)[..., 0]):
+        probabilities = torch.softmax(probabilities, 1)
+    return probabilities
```

Comment:
we already have this in heuristics, should we put it in utils for the usage of both?